src/core/ddsi/src/ddsi_plist.c

```diff
@@ -2049,6 +2049,7 @@ static const struct piddesc piddesc_omg[] = {
 #endif
   PP  (PROTOCOL_VERSION,                    protocol_version, Xox2),
   PP  (VENDORID,                            vendorid, Xox2),
+  PP  (PRODUCT_VERSION,                     product_version, Xox2),
```

Comment:
Almost right! `Xox2` means `o` times 2, so two octets. You really want four, which means `Xox4` but that one doesn't exist. It could be added, but it would be equivalent to `Xox2, Xox2`, or even to `Xo, Xo, Xo, Xo`. I probably would go for the latter (out of laziness? who knows ...)

This would be a vendor-specific code, and so it would need to be moved to the vendor-specific code tables, though. And as I remarked before, I think I would first see if it wouldn't make sense to clean up the `adlink_participant_info` in a (sufficiently) backwards compatible manner.